### ISSUE TITLE
prometheus target

### ISSUE THREAD

**Issue by ronendayan** [cloudworkz/kafka-minion#3]
How do i set prometheus target for the exporter ?  
For some reason when i set the prometheus, every time prometheus try to connect to the exporter, the exporter container is restart

**Comment by ronendayan** [comment]
I tried to test with some curl commands, getting:  
```
{"count":50,"level":"info","msg":"Starting '50' partition consumers","time":"2019-06-11T11:32:24Z","topic":"__consumer_offsets"}
{"count":50,"level":"info","msg":"Spawned all consumers","time":"2019-06-11T11:32:24Z","topic":"__consumer_offsets"}
{"level":"info","msg":"Listening on: '0.0.0.0:8080","time":"2019-06-11T11:32:24Z"}
{"level":"info","module":"storage","msg":"Registered 50 __consumer_offsets partitions which have to be consumed before metrics can be exposed","time":"2019-06-11T11:32:24Z"}
{"level":"info","msg":"Offets topic has not yet been consumed until the end","time":"2019-06-11T11:32:33Z"}
{"level":"info","module":"storage","msg":"Offset topic has been consumed","time":"2019-06-11T11:32:36Z"}
panic: runtime error: slice bounds out of range
goroutine 392 [running]:
github.com/google-cloud-tools/kafka-minion/collector.parseVersion(0xca1cb0, 0x1, 0xca1cb0, 0x1, 0xffffffffffffffff, 0xca1cb0, 0x1, 0x7fbac39b0d80)
/app/collector/collector.go:358 +0x227
github.com/google-cloud-tools/kafka-minion/collector.parseVersion(0xca1cb0, 0x1, 0x0, 0x0, 0x0, 0xcd0f30, 0xc000645000, 0xc000596de0)
/app/collector/collector.go:379 +0x1cc
github.com/google-cloud-tools/kafka-minion/collector.parseConsumerGroupName(0xca1cb0, 0x1, 0xc000596f50)
/app/collector/collector.go:346 +0x4a
github.com/google-cloud-tools/kafka-minion/collector.getVersionedConsumerGroups(0xc0005b25d0, 0xa)
/app/collector/collector.go:321 +0x197
github.com/google-cloud-tools/kafka-minion/collector.(*Collector).collectConsumerOffsets(0xc0004ae920, 0xc000312d20, 0xc0005b25d0, 0xc0005b2600, 0xc0005b2630)
/app/collector/collector.go:190 +0xb2
github.com/google-cloud-tools/kafka-minion/collector.(*Collector).Collect(0xc0004ae920, 0xc000312d20)
/app/collector/collector.go:130 +0x159
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
/go/pkg/mod/github.com/prometheus/client_golang@v0.9.2/prometheus/registry.go:434 +0x193
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
/go/pkg/mod/github.com/prometheus/client_golang@v0.9.2/prometheus/registry.go:526 +0xe23

```

**Comment by weeco** [comment]
Hey,
Kafka Minion should not crash when prometheus scrapes the target. I can see where this is happening, but I am not sure yet which input is causing this. Kafka Minion parses your consumer group id and tries to recognize a "version" in it (see readme how and why it does that).

Can you give me a list of all your consumer group names? One name should cause this panic.

**Comment by weeco** [comment]
@ronendayan I took another look at the function which caused the panic and I figured out if your consumer group name only consists of digits it would cause this panic. I pushed a fix and it'll be released along with another important fix under v0.1.3
